packages/utils/src/commonwell/cq-org-builder.ts

```diff
@@ -0,0 +1,176 @@
+  const flatOrgs: CQOrgHydrated[] = [];
+  for (const gateway of gatewaysToProcess) {
+    const orgs: OrgWithGateway[] = gateway.Organizations.map(o => {
+      return { id: o.Id, name: o.Name, gateway: gateway.Name };
+    });
+    const res = await executeAsynchronously(
+      orgs,
+      async org => {
+        const states: string[] = isLocal
+          ? await getStatesFromCurrent(org.id)
+          : await getOrgStatesFromSequoia(org.id);
+        flatOrgs.push({ ...org, states: states, prio: getPrio(org) });
+      },
+      { numberOfParallelExecutions: 100, keepExecutingOnError: true }
+    );
+    const failed = res.flatMap(r => (r.status === "rejected" ? r.reason : []));
+    const success = res.flatMap(r => (r.status === "fulfilled" ? "ok" : []));
+    console.log(
+      `--------------> Done w/ orgs of GW ${gateway.Name}: ${success.length} succeeded, ${failed.length} failed`
+    );
+    if (failed.length) {
+      fs.appendFileSync(outputFailedFilename, JSON.stringify(failed, null, 2));
+      const errorsOnThisGW = uniqBy(failed, f => f.reason);
+      totalFailed.push(...errorsOnThisGW.map(f => ({ gateway: gateway.Name, reason: f.reason })));
+    }
+  }
+  fs.writeFileSync(resultFilename, JSON.stringify(flatOrgs, null, 2));
+
+  console.log(`Saved flat org list to ${resultFilename}`);
+
+  console.log(`Failed`, totalFailed);
+
+  console.log(`Done at ${new Date().toISOString()}`);
+}
+
+async function getInfo(url: string) {
+  return retryFunction(() => axios.get(url), 5, 500);
+}
+
+async function getOrgStatesFromSequoia(orgOID: string): Promise<string[]> {
+  if (!orgOID) return [];
+  const url =
+    `${sequoiaQueryURL}` +
+    `?_format=json` +
+    `&_id=${orgOID}` +
+    `&_radius=30` +
+    `&_sort=orgname` +
+    `&_active=true` +
+    `&apikey=${sequoiaApiKey}`;
+
+  const resp = await getInfo(url);
+  if (!resp) return [];
+
+  const parsed: { name: string; state: string | undefined }[] = resp.data.Bundle?.entry?.map(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (e: any) => ({
+      name: e.resource?.Organization?.name?.value ?? uuidv4(),
+      state: e.resource?.Organization?.address?.state?.value,
+    })
+  );
+  const states: string[] = [];
+  for (const p of parsed) {
+    if (p.state) states.push(p.state);
+  }
+  console.log(`Got these states for org ${orgOID}: ${states.join(", ")}`);
+  return states;
+}
+
+async function getStatesFromCurrent(orgOID: string): Promise<string[]> {
+  const states = getOrgs().find(o => o.id === orgOID)?.states ?? [];
+  console.log(`(local) Got these states for org ${orgOID}: ${states.join(", ")}`);
+  return states;
+}
+
+function getPrio(org: OrgWithGateway): OrgPrio {
+  if (priorityOrgs.high.includes(org.id)) return "high";
+  if (priorityOrgs.medium.includes(org.id)) return "medium";
+  return "low";
+}
+
+// from API
+const retryFunction = async <K>(fn: () => Promise<K>, maxRetries = 3, waitTime = 3000) => {
```

Comment:
seems like this could be useful to have on some shared location

Reply:
yeah, this comes from the API, moved to shared